packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -51,6 +55,24 @@ export function errorHandling(errorDetails: any, label?: string, moreInfo?: stri
                 label = label.substring(0, label.indexOf(" ["));
             }
 
+            if (errorDetails.mDetails.additionalDetails) {
+                const tokenError: string = errorDetails.mDetails.additionalDetails;
+                if (tokenError.includes("Token is not valid or expired.")) {
+                    if (isTheia()) {
+                        vscode.window.showErrorMessage(errToken).then(async () => {
+                            await Profiles.getInstance().ssoLogin(null, label);
+                        });
+                    } else {
+                        vscode.window.showErrorMessage(errToken, "Log in to API Mediation Layer").then(async (selection) => {
```

Comment:
same here with error being more generalized to work with other profiles that don't authenticate with the API Mediation Layer.